File: backend/Models/Entities/Schedule.cs
using MedicalDemo.Enums;

namespace MedicalDemo.Models.Entities
{
    public partial class Schedule
    {
        public Schedule()
        {
            Dates = new HashSet<Date>();
            SwapRequests = new HashSet<SwapRequest>();
        }

        public Guid ScheduleId { get; set; }
        public ScheduleStatus Status { get; set; }
        public int GeneratedYear { get; set; }
        public Semester Semester { get; set; }

        public virtual ICollection<Date> Dates { get; set; }
        public virtual ICollection<SwapRequest> SwapRequests { get; set; }
    }
}

File: frontend/src/lib/models/Schedule.ts
export interface ScheduleStatus {
  id: number;
  description: string;
}

export interface ScheduleSemester {
  id: number;
  name: string;
}

export interface Schedule {
  scheduleId: string;
  status: ScheduleStatus;
  year: number;
  semester: ScheduleSemester;
}

export interface SchedulesByYear {
  [year: number]: Schedule[];
}